Guidelines/styleGuide.md

```diff
@@ -0,0 +1,47 @@
+## general
+### docker
+Since kubernetes is the agreed container managment framework, use docker images for runtimes.
+
+### object oriented design principles (recommendation)
+General best practices should be applied to software development.
+* SOLID: https://scotch.io/bar-talk/s-o-l-i-d-the-first-five-principles-of-object-oriented-design
+* DRY: https://de.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself
+
+### test driven development (recommendation)
+Use tdd (https://en.wikipedia.org/wiki/Test-driven_development)
+
+### clean code / clean architecture (recommendation)
+* https://gist.github.com/ygrenzinger/14812a56b9221c9feca0b3621518635b#file-cleanarchitecture-md
+* https://8thlight.com/blog/uncle-bob/2012/08/13/the-clean-architecture.html
+
+
+## javascript
+### node.js docker
+Use node.js docker images based on LTS version (currently node:carbon-alpine, node:carbon) instead of custom from scratch built containers.
```

Comment:
Should we enforce usage of yarn instead of npm?
As far as I see in code, yarn is main tool for dependency management in monorepo.
And as far as I understand this removes package-lock.json files from services and there would be only one yarn.lock for whole project.
But this makes docker build more complex  as yarn.lock is above service folder in folders hierarchy.

Reply:
We planned to use lerna for monorepo workspace support. If this is feasable, I dont see any advantage in yarn over npm.
If we choose lerna, I vote for using npm instead of yarn.

Reply:
We use yarn workspace in combination with lerna. Which is common practice in mono Repos.